### PR TITLE
ENH: make s3 client configurable for dataset hashing

### PR DESCRIPTION
**Pull request by mbaak**

### CODE REVIEW

File: tests/resources/tliba/src/tliba/random/rng.py [hunk 1]
```diff
@@ -14,7 +14,4 @@ def draw_normal_samples():
 
 def draw_bernoulli_samples():
     """Create table consisting of 5000 draws from a bernoulli distribution where the probability is drawn form a Beta(2, 5)."""
-    return pd.DataFrame(
-        np.random.binomial(1, draw_beta_samples().values.flatten(), 5000),
-        columns=["binomial"],
-    )
+    return pd.DataFrame(np.random.binomial(1, draw_beta_samples().values.flatten(), 5000), columns=["binomial"])
```
Comment (RUrlus) [inline tests/resources/tliba/src/tliba/random/rng.py:17]: Can we avoid any style changes in the tests directory? The module hash tests are failing because of this change

File: src/pycodehash/datasets/boto3.py
Comment (RUrlus) [inline src/pycodehash/datasets/boto3.py:1]: Could you rename this to `_s3_client.py` to mark it as an internal utility module rather than a dataset

File: src/pycodehash/datasets/__init__.py
Comment (RUrlus) [inline src/pycodehash/datasets/__init__.py:1]: Was this export missing previously? Because I don't see any new code that is related?

File: pyproject.toml
Comment (RUrlus) [inline pyproject.toml:1]: Can we move out these changes to their own commit? `PKG: Update ruff configuration`

File: tests/resources/tliba/src/tliba/random/rng.py (hunk 1 above)
Comment (mbaak) [reply tests/resources/tliba/src/tliba/random/rng.py]: Removed all style updates in test/

File: pyproject.toml
Comment (mbaak) [reply pyproject.toml:1]: Done, now in a separate commit.

File: src/pycodehash/datasets/boto3.py
Comment (mbaak) [reply src/pycodehash/datasets/boto3.py:1]: Done.

File: src/pycodehash/datasets/__init__.py
Comment (mbaak) [reply src/pycodehash/datasets/__init__.py:1]: Not sure, but this commit is removed now.
